### DISCUSSION TITLE
Is there a programmatic way to get the EKS cluster token using java aws sdk

### DISCUSSION THREAD

**Discussion by srinivasev** [aws/aws-sdk-java#2737] [Q&A] [answered]
I want to get the EKS cluster token programmatically using java aws sdk code. I couldn't find it in the samples code section. Below details are available with me for generating the token.

`aws_access_key_id, aws_secret_access_key, aws_session_token, region, cluster-name`

Kindly help,

**Comment by debora-ito** [comment, marked as answer]
There's no straightforward way to obtain the cluster authentication token using the Java SDK, I'm afraid. 
The [EKS Dev Guide](https://docs.aws.amazon.com/eks/latest/userguide/create-kubeconfig.html) points to the use of a aws-cli command : `aws eks get-token` or the [aws-iam-authenticator](https://github.com/kubernetes-sigs/aws-iam-authenticator) library.

We do have a feature request to add support in the Java SDK, and people have shared their working code:  https://github.com/aws/aws-sdk-java/issues/1795.
